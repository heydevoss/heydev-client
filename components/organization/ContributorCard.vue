<template>
  <div @click="redirectToUserPage">
    <el-row>
      <el-col>
        <el-card class="card" shadow="hover" :body-style="{ padding: '0px' }">
          <br /><img :src="contributor.avatarUrl" class="image" />
          <div style="padding: 0px;">
            <span>{{ contributor.login }}</span>

            <div class="description">
              <br /><span>{{ contributor.bio }}</span
              ><br />
              <br />
            </div>
          </div>
          <footer class="box-footer">
            <table>
              <tr>
                <td>{{ contributor.totalIssues }}</td>
                <td>{{ contributor.totalPullRequests }}</td>
                <td>{{ contributor.totalCommits }}</td>
              </tr>
              <tr>
                <td>Issues</td>
                <td>Pull Requests</td>
                <td>Commit</td>
              </tr>
            </table>
          </footer>
        </el-card>
      </el-col>
    </el-row>
  </div>
</template>

<script>
export default {
  name: 'ContributorCard',
  props: {
    contributor: Object
  },
  methods: {
    redirectToUserPage() {
      this.$router.replace(`/user/${this.contributor.login}`)
    }
  }
}
</script>

<style lang="scss" scoped>
table {
  border-collapse: separate;
  border-spacing: 13px;
  margin: 0 0 0 0;
}
.box-footer {
  position: absolute;
  align-self: center;
  bottom: 0;
}
.card {
  width: 250px;
  height: 350px;
  background-color: #47525e;
  color: white;
  cursor: pointer;
}

.description {
  text-align: center;
  font-size: 0.8rem;
  justify-content: center;
  margin: 3px;
}

.bottom {
  margin-top: 13px;
  line-height: 12px;
}

.button {
  padding: 0;
  float: right;
}

.image {
  border-radius: 100%;
  width: 100px;
  height: 100px;
  display: relative;
  position: center;
}
</style>
