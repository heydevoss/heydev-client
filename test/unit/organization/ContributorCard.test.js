import { shallowMount } from '@vue/test-utils'
import ContributorCard from '@/components/organization/ContributorCard.vue'

describe('<ContributorCard/>', () => {
  const wrapper = shallowMount(ContributorCard, {
    propsData: {
      contributor: {
        avatarUrl: '~/assets/avatar/default.png',
        login: 'test',
        bio: 'hey thats my test',
        totalIssues: 10,
        totalPullRequests: 2,
        totalCommits: 3
      }
    },
    stubs: {
      'el-card': true,
      'el-row': true,
      'el-col': true
    }
  })

  test('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('renders the table ', () => {
    expect(wrapper.html()).toContain('<table>')
  })

  it('should have the contributor login ', () => {
    expect(wrapper.html()).toContain('test')
  })

  it('should have the contributor bio ', () => {
    expect(wrapper.html()).toContain('hey thats my test')
  })

  it('should render avatar', () => {
    expect(wrapper.html()).toContain('<img')
  })

  test('should have the labels info (Issues)', () => {
    expect(wrapper.html().includes('Issues')).toBe(true)
    expect(wrapper.html().includes('10')).toBe(true)
  })

  test('should have the labels info (Commit)', () => {
    expect(wrapper.html().includes('Commit')).toBe(true)
    expect(wrapper.html().includes('3')).toBe(true)
  })

  test('should have the labels info (Pull Requests)', () => {
    expect(wrapper.html().includes('Pull Requests')).toBe(true)
    expect(wrapper.html().includes('2')).toBe(true)
  })
})
