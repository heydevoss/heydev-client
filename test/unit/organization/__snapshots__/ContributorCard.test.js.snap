// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContributorCard/> is a Vue instance 1`] = `
<div>
  <el-row-stub>
    <el-col-stub>
      <el-card-stub
        body-style="[object Object]"
        class="card"
        shadow="hover"
      >
        <br />
        <img
          class="image"
          src="~/assets/avatar/default.png"
        />
         
        <div
          style="padding: 0px;"
        >
          <span>
            test
          </span>
           
          <div
            class="description"
          >
            <br />
            <span>
              hey thats my test
            </span>
            <br />
             
            <br />
          </div>
        </div>
         
        <footer
          class="box-footer"
        >
          <table>
            <tr>
              <td>
                10
              </td>
               
              <td>
                2
              </td>
               
              <td>
                3
              </td>
            </tr>
             
            <tr>
              <td>
                Issues
              </td>
               
              <td>
                Pull Requests
              </td>
               
              <td>
                Commit
              </td>
            </tr>
          </table>
        </footer>
      </el-card-stub>
    </el-col-stub>
  </el-row-stub>
</div>
`;
